lib/multi_xml/parsers/rexml.rb

```diff
@@ -56,8 +58,7 @@ def collapse(element)
       def merge_texts!(hash, element)
         if element.has_text?
           # must use value to prevent double-escaping
-          texts = ''
-          element.texts.each { |t| texts << t.value }
+          texts = element.texts.reduce('') { |acc, t| acc + t.value }
```

Comment:
I believe this line will also mutate `''`, the initial value of acc. You'd need to use:
```ruby
texts = element.texts.reduce(+'') { |acc, t| acc + t.value }
```
alternatively:
```ruby
texts = element.texts.map(&:value).join
```